File: VeriJSONTests/Helpers/NSBundle+JSONLoader.m
//  Copyright (c) 2012 Daniel Cutting. All rights reserved.

#import "NSBundle+JSONLoader.h"

@implementation NSBundle (JSONLoader)

- (id)jsonFromResource:(NSString *)resource {
    NSString *path = [self pathForResource:resource ofType:nil];
    NSData *jsonData = [NSData dataWithContentsOfFile:path];
    id json = [NSJSONSerialization JSONObjectWithData:jsonData options:0 error:nil];
    if (!json) {
        @throw @"Could not load JSON.";
    }
    return json;
}

@end

File: VeriJSONTests/Helpers/NSBundle+JSONLoader.swift
//
//  NSBundle+JSONLoader.swift
//  VeriJSON-Swift
//
//  Created by Du, Xiaochen (Harry) on 3/5/15.
//  Copyright (c) 2015 Where Here. All rights reserved.
//

import Foundation

extension NSBundle {
    
    func jsonFromResource(resource:String) -> AnyObject? {
        
        if let path = self.pathForResource(resource, ofType: nil) {
            if let jsonData = NSData(contentsOfFile: path) {
                return try? NSJSONSerialization.JSONObjectWithData(jsonData, options: NSJSONReadingOptions.MutableContainers)
            }
        }
        return nil;
        

    }
}